common/src/main/java/com/wynntils/utils/wynn/InventoryUtils.java

```diff
@@ -54,6 +61,26 @@ public static ItemStack getItemInHand() {
         return McUtils.player().getItemInHand(InteractionHand.MAIN_HAND);
     }
 
+    /**
+     * Checks if the weapon in the player's hand has all requirements met.
+     */
+    public static boolean itemRequirementsMet(ItemStack itemStack) {
+        Optional<WynnItem> wynnItem = Models.Item.getWynnItem(itemStack);
+
+        if (wynnItem.isEmpty()
+                || !(wynnItem.get() instanceof GearItem) && !(wynnItem.get() instanceof CraftedGearItem)) {
+            return false;
+        }
+
+        for (StyledText loreLine : LoreUtils.getLore(itemStack)) {
+            if (loreLine.startsWith("§c✖") && loreLine.contains("Min: ")) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
     public enum MouseClickType {
```

Comment:
This method does not fit this util class. But I don't think it should be moved, instead you should be able to to make GearItem/CraftedGearItem able to tell us if all requirements are met. Perhaps the start would be creating a `RequirementItemProperty`.